number_theory/prime_unittest.cpp

```diff
@@ -83,4 +85,32 @@ TEST(EulerSieveTest, MinPrimeFactor) {
   test_min_prime_factor<uint64_t>();
 }
 
+template <typename T>
+void test_coprime_pairs() {
+  T n = 100;
+  auto pairs = coprime_pairs(n);
+  for (auto [x, y] : pairs) {
+    ASSERT_GE(n, x);
+    ASSERT_GE(x, y);
+    ASSERT_GE(y, T(0));
+  }
+  std::set<std::pair<T, T>> pairs_set(pairs.begin(), pairs.end());
+  for (T x = 0; x <= n; ++x)
+    for (T y = 0; y <= x; ++y) {
+      bool is_coprime = std::gcd(x, y) == 1;
```

Comment:
Wouldn't it be faster if we compared the result with an expected result just like we did it for sieves? This also looks nice to be honest, I am just reminding the other option.

Reply:
The result list is too long to be hard-coded. Actually O(n^2 log n) is fast enough since n is 100.